### DISCUSSION TITLE
Text interline spacing with pango

### DISCUSSION THREAD

**Discussion by tgd-git** [ImageMagick/ImageMagick#3827] [Help]
Is there any way to set the interline spacing when using the pango coder for text rendering?

I have found this old topic on the forums where a guy patched this option and built from source.
https://legacy.imagemagick.org/discourse-server/viewtopic.php?p=168449#p168449
I hoped in the meanwhile this maybe got updated in the public binaries but seems not.

I tried to set " -interline-spacing 33 " with the convert command and also as " -define pango:spacing=33 " as well as defining spacing in the markup. Nothing worked :(

I need to use pango to correctly render some arabic text and line spacing is pretty important.

**Comment by snibgo** [comment]
> Is there any way to set the interline spacing when using the pango coder for text rendering?

As far as I can see: no, other than applying a patch such as shown in your link.

The patch would solve your problem, and may be useful to others, so the developers might incorporate this. Using a new define "pango:spacing" won't break any existing scripts. Using IM's "-interline-spacing" for the value seems more logical, but might break some scripts.

**Reply by tgd-git** [reply]
I am on windows and I have never built IM from source. I am super scared of the build or even setting up a dev environment for this. Pango/Cairo on windows seems a big pain too from what i have read so far.

**Comment by snibgo** [comment]
Pango is a much simpler version of HTML. For many purposes, it is too simple. I use wkhtmltoimage.exe to rasterize local HTML pages, but I expect other software also works.

**Comment by dlemstra** [comment]
I will take a look at this and check if we could add this to ImageMagick 7. It would help me if I had a sample command that I could use to test this.

**Reply by tgd-git** [reply]
This should be the very basic command:

`convert.exe pango:@mytext.txt mypng.png`

`mytext.txt`:

```
<markup><span>Lorem ipsum dolor sit amet, 
consectetur adipiscing elit, 
sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua. </span></markup>
```
expected output `mypng.png`:

![image](https://user-images.githubusercontent.com/2562637/122134563-22f9f480-ce3f-11eb-99ca-d385d61f33e8.png)

**Comment by dlemstra** [comment]
Support for this is available in the latest release of ImageMagick.
